shared/actions/teams/index.js

```diff
@@ -26,14 +26,17 @@ import {convertToError} from '../../util/errors'
 import type {TypedState} from '../../constants/reducer'
 
 const _createNewTeam = function*(action: Constants.CreateNewTeam) {
-  const {payload: {name}} = action
+  const {payload: {name, routePath}} = action
   yield Saga.put(Creators.setTeamCreationError(''))
   yield Saga.put(Creators.setTeamCreationPending(true))
   try {
     yield Saga.call(RPCTypes.teamsTeamCreateRpcPromise, {
       param: {name, sendChatNotification: true},
     })
 
+    // Dismiss the create team dialog.
+    yield Saga.put(navigateTo([], routePath.butLast()))
+
```

Comment:
Or I could just change the below to navigate to the teamsTab unconditionally.

Reply:
We should go to absolute paths and not relative ones I think. Additionally if this happens as part of a side effect (like here) I'd use the action of on path helper

Reply:
Hmm, after playing around with this for a while, I think there's nothing that currently does what I want, which is, roughly, if the subpath starting from a given single-node path is the same as a given one, navigate to another subpath. In this case, if the subpath starting from [teams] is the same as [...parentPath, newTeamDialog], navigate to [...parentPath].

`putActionIfOnPath` doesn't work if you navigate away from the teams tab before the team creation completes -- it won't take effect and therefore if you switch back to the teams tab, it'll still be on newTeamDialog.

However, the current way doesn't work either, because if you navigate to a different teams page, it'll navigate to parentPath instead.

Probably need another helper for this. I'll see what I can do on Monday.

Reply:
Ah, turns out `putActionifOnPath` takes a parentPath argument, which seems to be what I want!